File: app/db/models/workflow.py
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column

from app.db.session import Base


class WorkflowDefinitionModel(Base):
    """
    Database model for WorkflowDefinition.
    """

    __tablename__ = "workflow_definitions"

    uuid: Mapped[str] = mapped_column(primary_key=True, index=True)
    id: Mapped[str | None]
    name: Mapped[str]
    description: Mapped[str | None]
    steps: Mapped[list[dict]] = mapped_column(JSONB)

File: rust-app/src/db/models/workflow.rs
use sqlx::FromRow;

/// Database model for WorkflowDefinition.
///
/// Mirrors Python's WorkflowDefinitionModel from app/db/models/workflow.py.
#[derive(Debug, Clone, FromRow)]
pub struct WorkflowDefinitionRow {
    pub uuid: String,
    pub id: Option<String>,
    pub name: String,
    pub description: Option<String>,
    pub steps: serde_json::Value,
}
